### DISCUSSION TITLE
Getting data out - API

### DISCUSSION THREAD

**Discussion by SupraJames** [OpnForm/OpnForm#564] [Q&A]
I am trying to figure out the best way of getting responses out of a form in an automated manner.

The use case is for a club where members need to self log activities when the club is not manned. So maybe 20 or so responses in a day. The next day, the club admin would need to grab those reponses and insert them into a spreadsheet. I do not want to ask them to go through 20 emails and extract the data that way!

Ideally, I'd like there to be a daily .CSV sent at the end of each day to a mailbox with all the responses from that day.

That functionality doesn't seem to exist, so I thought about using the API. I created an access token, but functionality seems really limited when compared to the "real" API. 

The endpoint `external/zapier/submissions/recent?form_id=1` only seems to return the one latest response.

Open to ideas here, any help would be appreciated!

**Comment by JhumanJ** [comment]
Hi there! Currently the only way is via webhooks (or other integrations). We don't have yet a public API to fetch existing submissions. If you use google sheets we also have an integration available in the cloud version!

**Comment by JhumanJ** [comment]
We now have an API! https://docs.opnform.com/api-reference/introduction
